src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/RuntimeInformation.Windows.cs

```diff
@@ -31,25 +31,44 @@ public static string OSDescription
             }
         }
 
-        public static Architecture OSArchitecture
+        public static unsafe Architecture OSArchitecture
         {
             get
             {
                 int osArch = s_osArchPlusOne - 1;
 
                 if (osArch < 0)
                 {
-                    Interop.Kernel32.SYSTEM_INFO sysInfo;
-                    unsafe
+                    // If we are running an x64 process on a non-x64 windows machine, we will report x64 as OS architecutre.
+                    //
+                    // IsWow64Process2 is only available on Windows 10+, so we will perform run-time introspection via indirect load
+                    if (NativeLibrary.TryGetExport(NativeLibrary.Load(Interop.Libraries.Kernel32), "IsWow64Process2", out IntPtr isWow64Process2Ptr))
                     {
+                        ushort processMachine, nativeMachine;
+                        var isWow64Process2 = (delegate* unmanaged<IntPtr, ushort*, ushort*, int>)isWow64Process2Ptr;
+                        if (isWow64Process2(Interop.Kernel32.GetCurrentProcess(), &processMachine, &nativeMachine) != 0)
+                        {
+                            osArch = (int)MapMachineConstant(nativeMachine);
+                        }
+                        else
+                        {
+                            Debug.Fail("Call to IsWow64Process2() failed unexpectedly.");
+                        }
+                    }
+
+                    if (osArch == -1)
```

Comment:
Why are we be checking `if (osArch < 0)` above and `osArch >= 0 ` below, but `if (osArch == -1)` here instead? - I would think if this is essentially a fallback, it should check `osArch < 0` again.